code/game/turfs/turf.dm

```diff
@@ -191,16 +191,22 @@
 	// First, make sure it can leave its square
 	if(isturf(mover.loc))
 		// Nothing but border objects stop you from leaving a tile, only one loop is needed
-		for(var/obj/obstacle in mover.loc)
-			if(!obstacle.CheckExit(mover, src) && obstacle != mover && obstacle != forget)
+		// This as anything looks odd, since we check istype shortly after, but it's so we can save an istype check for items, which are far more common than other objects.
+		for(var/obj/obstacle as anything in mover.loc)
```

Comment:
You're continuing on !istype, any reason to do `as anything` in this case?

Reply:
See the comment above.